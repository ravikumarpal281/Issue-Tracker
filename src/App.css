.signup-form {
  width: 100%;
  max-width: 330px;
  padding: 15px;
  margin: auto;
  height: 100%;
}

.card {
  border-radius: 4px;
  background: #292b2c;
  box-shadow: 0 6px 10px rgba(0, 0, 0, 0.08), 0 0 6px rgba(0, 0, 0, 0.05);
  transition: 0.3s transform cubic-bezier(0.155, 1.105, 0.295, 1.12),
    0.3s box-shadow,
    0.3s -webkit-transform cubic-bezier(0.155, 1.105, 0.295, 1.12);
  overflow: hidden;
}

.card:hover {
  transform: scale(1.1);
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.12), 0 4px 8px rgba(0, 0, 0, 0.06);
  overflow: hidden;
}

.text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.main-nav-active {
  color: tomato;
}

.main-nav:hover {
  color: tomato;
}

.sign-in-form {
  background-color: lightsteelblue;
  display: flex;
  margin-top: 1px;
  width: 100%;
}

.sign-in-form > div {
  font-size: 20px;
  width: 100%;
}

.form-style {
  margin-left: 10px;
  margin-bottom: 10px;
  margin-right: 10px;
  align-items: center;
}
