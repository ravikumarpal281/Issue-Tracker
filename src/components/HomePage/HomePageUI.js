import React from "react";
import IssuesList from "../IssuesList";
import { Spinner } from "react-bootstrap";

const HomePageUI = ({ issuesData }) => {
  return issuesData.loading ? (
    <Spinner animation="border" variant="success" />
  ) : issuesData.error ? (
    <h2>{issuesData.error}</h2>
  ) : (
    <div>
      <IssuesList issues={issuesData.issues}></IssuesList>
    </div>
  );
};

export default HomePageUI;
