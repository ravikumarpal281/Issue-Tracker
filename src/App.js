import "bootstrap/dist/css/bootstrap.min.css";
import { Provider } from "react-redux";
import "./App.css";
import HomePage from "./components/HomePage";
import NavBar from "./components/NavBar";
import store from "./components/redux/store";
import SignUp from "./components/SignUp";
import SignIn from "./components/SignIn";
function App() {
  return (
    <Provider store={store}>
      <div className="App">
        <NavBar></NavBar>
        <SignIn></SignIn>
        {/* <SignUp></SignUp> */}
        {/* <HomePage></HomePage> */}
      </div>
    </Provider>
  );
}

export default App;
