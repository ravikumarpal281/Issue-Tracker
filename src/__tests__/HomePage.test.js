import React from "react";
import ReactDOM from "react-dom";
import toJson from "enzyme-to-json";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "../App";
import HomePageUI from "../components/HomePage/HomePageUI";
import { Spinner } from "react-bootstrap";

Enzyme.configure({ adapter: new Adapter() });

it("Home Page Snap", () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("Error page test", () => {
  const issuesData = {
    loading: false,
    error: "This is error",
    issues: [],
  };
  const wrapper = shallow(<HomePageUI issuesData={issuesData} />);
  expect(wrapper.find("h2").text()).toEqual("This is error");
});

it("Props test in Home Page", () => {
  const issuesData = {
    loading: false,
    error: "",
    issues: [],
  };
  const wrapper = shallow(<HomePageUI issuesData={issuesData} />);
  expect(wrapper.find("div").length).toEqual(1);
});

it("Spinner test in Home Page", () => {
  const issuesData = {
    loading: true,
    error: "",
    issues: [],
  };
  const wrapper = shallow(<HomePageUI issuesData={issuesData} />);
  expect(wrapper.find(Spinner).length).toEqual(1);
});
