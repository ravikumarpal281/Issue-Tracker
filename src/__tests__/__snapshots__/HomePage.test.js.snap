// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Snap 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div
      className="App"
    >
      <withRouter(Navigation) />
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/AddIssue"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/Login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "type": [Function],
            }
          }
          exact={true}
          path="/Register"
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/issue/:id"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/editIssue/:id"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/TopIssues"
        />
      </Switch>
    </div>
  </BrowserRouter>
</Provider>
`;
